src/rgw/rgw_op.cc

```diff
@@ -4551,15 +4555,15 @@ void RGWCopyObj::execute()
 			   &mtime,
 			   mod_ptr,
 			   unmod_ptr,
-                           high_precision_time,
+         high_precision_time,
 			   if_match,
 			   if_nomatch,
 			   attrs_mod,
-                           copy_if_newer,
+         copy_if_newer,
 			   attrs, RGW_OBJ_CATEGORY_MAIN,
 			   olh_epoch,
 			   (delete_at ? *delete_at : real_time()),
-			   (version_id.empty() ? NULL : &version_id),
+			   &dst_version_id,
```

Comment:
what happens to the former NULL case?

Reply:
think version_id.empty() is enough, and it might be more proper to use string &version instead of string *version for RGWRados::copy_obj